workbench/src/main/setupAddPlugin.js

```diff
@@ -0,0 +1,58 @@
+import upath from 'upath';
+import fs from 'fs';
+import toml from 'toml';
+import { execSync } from 'child_process';
+import { app, ipcMain } from 'electron';
+
+import { getLogger } from './logger';
+import { ipcMainChannels } from './ipcMainChannels';
+import { settingsStore } from './settingsStore';
+
+const logger = getLogger(__filename.split('/').slice(-1)[0]);
+
+export default function setupAddPlugin() {
+  ipcMain.handle(
+    ipcMainChannels.ADD_PLUGIN,
+    (e, pluginURL) => {
+      logger.info('adding plugin at', pluginURL);
+      const pluginsDir = upath.join(app.getPath('userData'), 'plugins');
+
+      // Download the plugin from its remote git repo
+      if (!fs.existsSync(pluginsDir)) {
+        fs.mkdirSync(pluginsDir);
+      }
+      const pluginRepoName = upath.basename(pluginURL, upath.extname(pluginURL));
+      fs.rmSync(`${pluginsDir}/${pluginRepoName}`, { recursive: true, force: true });
+      execSync(`git clone ${pluginURL} "${pluginsDir}/${pluginRepoName}"`, { stdio: 'inherit' });
+
+      // Read in plugin metadata from the pyproject.toml
+      const pyprojectTOML = toml.parse(fs.readFileSync(
+        `${pluginsDir}/${pluginRepoName}/pyproject.toml`
+      ).toString());
+      const pluginID = pyprojectTOML.tool.natcap.invest.model_id;
+      const pluginName = pyprojectTOML.tool.natcap.invest.model_name;
+      const pluginPyName = pyprojectTOML.tool.natcap.invest.pyname;
+
+      // Create a conda env containing the plugin and its dependencies
+      const envName = `invest_plugin_${pluginID}`;
+      fs.writeFileSync('tmp_env.txt', 'python<3.12\ngdal<3.6');
```

Comment:
Not intended to be all of the plugin's dependencies. These are just the prerequisites to install `natcap.invest`. When we `pip install` the plugin, `pip` will install `natcap.invest` (because the plugin depends on it). And that will fail if `gdal` isn't already installed.

Eventually, we could define the plugin's conda dependencies in `pyproject.toml`, see [PEP 725](https://peps.python.org/pep-0725/)